src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Fx2Nl/Fx2NlLanguageServerOperationHandler.cs

```diff
@@ -88,18 +83,25 @@ private async Task Fx2NlAsync(LanguageServerOperationContext operationContext, C
         /// <param name="cancellationToken">Cancellation Token.</param>
         public async Task HandleAsync(LanguageServerOperationContext operationContext, CancellationToken cancellationToken)
         {
-            if (!operationContext.TryParseParamsAndAddErrorResponseIfNeeded(out _fx2NlRequestParams))
+            if (!operationContext.TryParseParamsAndAddErrorResponseIfNeeded(out CustomFx2NLParams fx2NlRequestParams))
             {
                 return;
             }
 
-            await PreHandleFx2NlAsync(operationContext, cancellationToken).ConfigureAwait(false);
-            if (_preHandleCheckResult == null)
+            var nlHandler = await operationContext.GetNLHandler(fx2NlRequestParams.TextDocument.Uri, _nLHandlerFactory, fx2NlRequestParams, cancellationToken).ConfigureAwait(false);
+            _ = nlHandler ?? throw new ArgumentNullException(nameof(nlHandler));

```

Comment:
This exception (***Argument** NullException for the nlHandler argument) is not appropriate (there is not an argument called nlHandler, and the error isn't about an argument being null). Possibly InvalidOperationException would be better here.

Reply:
actually nl handler is not expected to be null. this is over defensiveness at best (to myself :))